server/container_create_linux.go

```diff
@@ -440,39 +440,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		specgen.SetProcessNoNewPrivileges(securityContext.NoNewPrivs)
 
 		if !ctr.Privileged() {
-			for _, mp := range []string{
```

Comment:
I am not seeing any toggling of this behavior based on the security context

Reply:
There isn't really a way to do that, either Kubernetes provides the masked paths, or it doesn't.

In containerd they do have an option to disable this support, which they suggest enabling on Kubernetes <=1.11: https://github.com/containerd/containerd/blob/2b661b890f689716b2928408e386e656d8dd7118/pkg/cri/config/config.go#L283-L285

(The actual implementation in containerd is similar to this: https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create_linux.go#L202-L212)

I could add a similar option, but I don't believe cri-o aims to support Kubernetes versions that old?

Reply:
ah I see so kubelet unconditionally passes these down, unless the feature gate is on and the security context knob is set.